### PR TITLE
Fix race condition in options startup, and parsing empty JSON

### PR DESCRIPTION
**Pull request by nielm**
Fixes a race condition on startup where onDisplayChanged could get called before onPageLoad had finished populating the editors.

Also fixed an issue where if the contents of the editors was empty, you get an odd error message ("unexpected content")

And improved comments in options.js

### CODE REVIEW

File: window_manager/options.js [hunk 1]
```diff
@@ -413,13 +436,15 @@ function onDisplayChanged() {
...
 
 
-/** @return {Promise<void>} */
-function onSaveClick() {
-  return validateEverything()
-      .then((validatedConfig) => storage.save(validatedConfig))
-      .then(() => setStatus('Options saved'))
-      .then(() => undefined)
-      .catch((e) => setStatus(e.message));
+/** Save settings */
```
Comment (lukaszg84) [inline window_manager/options.js]: can you revert these lines? I would prefer promise.then... syntax as it is used in most of the codebase.

File: window_manager/options.js [hunk 2]
```diff
@@ -382,27 +395,37 @@ function getJsonTextFromEditor(editor) {
...
+  settingsEditor.set({text: config.settings});
+
+  // Make editors visible.
+  [...document.getElementsByClassName('jsonEditor')].forEach(
+      (/** @type {HTMLElement} */ e) => e.style.display='');
+  // Remove "Loading" messages.
+  [...document.getElementsByClassName('loading')].forEach(
+      (/** @type {HTMLElement} */ e) => e.style.display='none');
+
+  const validatedConfig = await validateEverything();
```
Comment (lukaszg84) [inline window_manager/options.js]: validateEverything will update the displays if needed. I think everything below this line can be removed.

File: window_manager/options.js [hunk 3]
```diff
@@ -382,27 +395,37 @@ function getJsonTextFromEditor(editor) {
 
 /**
  * Loads the config, and sets up editors.
- *
- * @return {Promise<void>}
  */
-function onPageLoad() {
-  actionsEditor= createJSONEditorForElement('actionsInput', [], schemaValidator.actions);
+async function onPageLoad() {
```
Comment (lukaszg84) [inline window_manager/options.js]: I think no changes are required to the onPageLoad - the only problem was that onLoad the onDisplayChanged method was called.
OnPageLoad is already calling validateEverything which updates displays.

Comment (nielm) [reply window_manager/options.js]: reverted. 
The only issue here is  that if the actions fails validation, then the displays list will not be populated until this validation error is fixed. 

(this also happens with onDisplayChanged() as well)

File: window_manager/options.js (hunk 2 above)
Comment (nielm) [reply window_manager/options.js]: reverted changes to function

File: window_manager/options.js (hunk 1 above)
Comment (nielm) [reply window_manager/options.js]: done

**Pull request merged**